javascript/testing/README.md

```diff
@@ -487,6 +491,65 @@ jest.mock('moment', ()=> ({
 }))
 ```
 
+## End-to-end Testing
+> For End-to-end testing we use [Cypress](https://www.cypress.io/) Framework. However, our suggested concepts could be adopted with other End-to-End frameworks/test runners. 
+
+The below diagram illustrates main blocks that involved in End-to-end test. 
+
+![End-to-end Test diagram](./resources/e2eDiagram.png)
+
+- **Test** 
+    - Uses the [AAA pattern](#aaa-pattern---arrange-act-assert)
+    - One and only one unit that __has assertions__ for testing components' integration and data integrity 
+    - Uses one or more `Page Objects` to access and interact with tested application page
+    
+Here is a simple example of test that checks search data functionality:    
+
+``` javascript
+import TestPageObject from './TestPageObject'
+const testPageObject =  new TestPageObject()
+
+it('should search data correctly', () => {
```

Comment:
I think that in this test we are actually checking that searchTerm was displayed and not that data was searched correctly.
Or we need to change the assertion (which should check that there are corrected filtered elements) or change the title of the test (which I prefer more because don't want to make an example more complicated)

Reply:
I think that it checks filter by search term functionality, since the last assertion is getSearchResults().should('contain', 'searchTerm') - that checks search functionality